be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -510,6 +514,18 @@ FileColumnIterator::FileColumnIterator(ColumnReader* reader) : _reader(reader) {
 Status FileColumnIterator::init(const ColumnIteratorOptions& opts) {
     _opts = opts;
     RETURN_IF_ERROR(get_block_compression_codec(_reader->get_compression(), _compress_codec));
+    if (config::enable_low_cardinality_optimize) {
+        auto dict_encoding_type = _reader->get_dict_encoding_type();
+        if (dict_encoding_type != ColumnReader::UNKNOWN_DICT_ENCODING) {
```

Comment:
better  if (xxx == xxx) {} else {}
if ( xxx != xxx)  is very hard to understand.